src/notes/notes.service.ts

```diff
@@ -113,6 +144,22 @@ export class NotesService {
     }
   }
 
+  private createGroupPermission(
+    note: Note,
+    group: Group,
+    permission: DefaultAccessPermission,
+  ): Optional<NoteGroupPermission> {
+    return Optional.ofNullable(permission)
```

Comment:
I think it would be more readable not to use functional pipeline and use instead a if-else-clause.